itou/static/js/utils.js

```diff
@@ -48,6 +48,28 @@ htmx.onLoad((target) => {
     $(target).attr("disabled", this.checked)
   })
 
+  /**
+  * JS to disable/enable and set another select field value.
+  **/
+  function toggle_disable_and_set_value(elt) {
+    const targetId = elt.getAttribute("data-disable-target");
+    const targetValue = elt.getAttribute("data-target-value");
+    const isSet = $(elt).val().length > 0;
+    if (isSet) {
+      $(targetId).val(targetValue);
+    }
+    $(targetId).attr("disabled", isSet);
+    $(targetId).attr("readonly", isSet);
+  }
+
+  const selectFieldWithDisable = target.querySelector('select[data-disable-target]');
+  if (selectFieldWithDisable.length > 0) {
+    toggle_disable_and_set_value(selectFieldWithDisable);
+    $(selectFieldWithDisable).change(function (e) {
+      toggle_disable_and_set_value(this);
+    })
+  }
+
```

Comment:
7 différences :
- utilisation d’une fonction nommée au lieu d’une fonction anonyme pour l’event listener. Ça permet de s’assurer que la fonction ne sera pas branchée deux fois.
- suppression du `selectFieldWithDisable.length > 0`, autant directement itérer, et ça marchera mieux s’il y a plusieurs éléments avec cette classe sur la page.
- suppression du `readonly` que je ne comprends pas (`disabled` devrait à la fois indiquer que le champ est désactiver et empêcher toute interaction, en plus de ne pas soumettre le champ).
- suppression de la variable inutile `targetValue` si `isSet === false`
- camelCase le nom de la fonction
```suggestion
  function toggleDisableAndSetValue() {
    const targetId = this.getAttribute("data-disable-target");
    const isSet = $(this).val().length > 0;
    if (isSet) {
      $(targetId).val(this.getAttribute("data-target-value"));
    }
    $(targetId).attr("disabled", isSet);
  }

  target.querySelectorAll('select[data-disable-target]').forEach(function(selectFieldWithDisable) {
    toggleDisableAndSetValue.call(selectFieldWithDisable);
    $(selectFieldWithDisable).change(toggleDisableAndSetValue);
  });

```

Je me demande si on ne veut pas aussi bloquer le sélecteur de commune si un pays autre que la France est sélectionné ?

Reply:
Merci ! :star: :star: (une pour le code et une pour les explications)

Reply:
> Je me demande si on ne veut pas aussi bloquer le sélecteur de commune si un pays autre que la France est sélectionné ?

Bonne idée, pour une autre PR ! :)